### DISCUSSION TITLE
I would like to pass the trace id to my logs on every message.

### DISCUSSION THREAD

**Discussion by rolfwessels** [serilog-contrib/serilog-sinks-grafana-loki#232] [Q&A]
First, thank you to the creators for this package. It's been handy.

Unfortunately, I would like to add TraceId to the labels but I do not seem to be able to do this

```csharp
// Serilog.Sinks.Grafana.Loki 8.2.0
builder.Host.UseSerilog((_, lc) =>
{
  lc.MinimumLevel.Override("Microsoft.AspNetCore", LogEventLevel.Warning)
    .Enrich.FromLogContext()
    .WriteTo.Console(
      outputTemplate: "{Timestamp:HH:mm:ss} {Level:u3}] {Message:lj}({TraceId:l}){NewLine}{Exception}")
    .WriteTo.GrafanaLoki(
      telemetrySettings.LokiUrl,
      labels: lokiLabels,
      propertiesAsLabels: PropertiesAsLabels(),
      credentials: FromUrl(telemetrySettings.LokiUrl),
      textFormatter: new MessageTemplateTextFormatter("{Message:lj} ({TraceId:l}){Exception}")
      );
});
```
Calling the following:
```cs
_log.Information("ScansSelect {trace}",Activity.Current?.TraceId);
```
The output to the console is as expected. 
```txt
ScansSelect 1b24ba06553df0257f4d9604085d872d(1b24ba06553df0257f4d9604085d872d)
```
But the loki output is

![image](https://github.com/serilog-contrib/serilog-sinks-grafana-loki/assets/643179/806fb6bb-9853-4047-8ce6-15dc4d549ccc)

Can someone please point me in the right direction?

Thanks

**Comment by mishamyte** [comment]
Hi @rolfwessels!

For your question I would suggest checking the source code of the [MessageTemplateTextFormatter (I highlighted the line)](https://github.com/serilog/serilog/blob/653260089620944516cbd874449f3c4bd02edc0e/src/Serilog/Formatting/Display/MessageTemplateTextFormatter.cs#L76).

Also I believe you could got a lot from the **structured** logging if you will use [LokiJsonTextFormatter and Grafana Loki json parser](https://github.com/serilog-contrib/serilog-sinks-grafana-loki/wiki/Using-LokiJsonTextFormatter-and-Grafana-Loki-json-parser). In that case you will get a lot of useful properties, that could be used for the work with logs.

If you wondering how it works, I would suggest checking the [Serilog.Sinks.Grafana.Loki.SampleWebApp](https://github.com/serilog-contrib/serilog-sinks-grafana-loki/tree/master/sample/Serilog.Sinks.Grafana.Loki.SampleWebApp) from that repository (you could spin up all required infra, using `docker compose` file in that repo).

The structured logging could give you more benefits, comparing to the simple `console-like` rendering style

**Reply by rolfwessels** [reply]
Thanks for the quick reply. I have tried json structure but that is also missing the traceId. The link to the [MessageTemplateTextFormatter (I highlighted the line)](https://github.com/serilog/serilog/blob/653260089620944516cbd874449f3c4bd02edc0e/src/Serilog/Formatting/Display/MessageTemplateTextFormatter.cs#L76)  seems to suggest that the trace property is missing on the log-event for Grafana.Loki. Is there an additional setting that I need to pass to "GrafanaLoki" options to enable that traceId for all logging events?

**Reply by mishamyte** [reply]
@rolfwessels you are right, I did a bit deeper investigation and find out that those properties were changes in Serilog's [v3.1.0](https://github.com/serilog/serilog/releases/tag/v3.1.0)

As for now we have a minimal version 2.12.0, it doesn't have those properties and they are lost during the copy process.

So I believe we will implement it in v9 (currently WIP).

I created the issue #233 for tracking of that feature.

**And thank you a lot for that founding!**

**Reply by mishamyte** [reply]
And as for you, if it is possible, I would suggest to downgrade all Serlog's deps to `3.1.0` or lower and just use `{TraceId}` alias.

So there are the changes to the `SampleWebApp` project:

```xml
 <ItemGroup>
        <PackageReference Include="Serilog.AspNetCore" Version="7.0.0" />
        <PackageReference Include="Serilog.Settings.Configuration" Version="7.0.1" />
    </ItemGroup>
```

```csharp
    [HttpGet("info")]
    public IActionResult GetInfo()
    {
        _logger.LogInformation("ScansSelect {TraceId}", Activity.Current?.TraceId);
        ...
```

And it worked as a charm for me

![image](https://github.com/serilog-contrib/serilog-sinks-grafana-loki/assets/5649656/4b263319-5a17-4e87-8bdd-cfe606c76d75)
